mis_builder/models/mis_builder.py

```diff
@@ -211,9 +212,8 @@ def _render_num(self, lang_id, value, divider,
             '%%%s.%df' % (sign, dp),
             value,
             grouping=True)
-        value = u'%s\N{NO-BREAK SPACE}%s%s' % \
-            (value, divider_label, suffix or '')
-        value = value.replace('-', u'\N{NON-BREAKING HYPHEN}')
+        value = u'%s%s\N{NO-BREAK SPACE}%s%s' % \
```

Comment:
I'd put a no-break space between prefix and value too (or perhaps a 'narrow no-break space' for both).